crates/api/src/lib.rs

```diff
@@ -49,29 +50,70 @@ pub use persistence::ConnectionOptions;
 use user_error::UFE;
 use uuid::Uuid;
 
-custom_error! {
-  pub ApiError
-    Store{source: persistence::StoreError}                      = "Storage {source:?}",
-    Transaction{source: diesel::result::Error}                  = "Transaction failed {source}",
-    Iri{source: iref::Error}                                    = "Invalid IRI {source}",
-    JsonLD{message: String}                                     = "Json LD processing {message}",
-    Ledger{source: SubmissionError}                             = "Ledger error {source}",
-    Signing{source: SignerError}                                = "Signing {source}",
-    NoCurrentAgent{}                                            = "No agent is currently in use, please call agent use or supply an agent in your call",
-    CannotFindAttachment{}                                      = "Cannot locate attachment file",
-    ApiShutdownRx                                               = "Api shut down before reply",
-    ApiShutdownTx{source: SendError<ApiSendWithReply>}          = "Api shut down before send",
-    LedgerShutdownTx{source: SendError<LedgerSendWithReply>}    = "Ledger shut down before send",
-    AddressParse{source: AddrParseError}                        = "Invalid socket address {source}",
-    ConnectionPool{source: r2d2::Error}                         = "Connection pool {source}",
-    FileUpload{source: std::io::Error}                          = "File upload",
-    Join{source: JoinError}                                     = "Blocking thread pool",
-    Subscription{source: SubscriptionError}                     = "State update subscription {source}",
-    NotCurrentActivity{}                                        = "No appropriate activity to end",
-    EvidenceSigning{source: common::k256::ecdsa::Error}         = "Could not sign message",
-    Contradiction{source: Contradiction}                        = "Contradiction {source}",
-    ProcessorError{source: ProcessorError}                      = "Processor {source}",
-    IdentityError{source: IdentityError}                        = "Identity {source}",
+#[derive(Error, Debug)]
+pub enum ApiError {
+    #[error("Storage: {0:?}")]
+    Store(#[from] persistence::StoreError),
+
+    #[error("Transaction failed: {0}")]
+    Transaction(#[from] diesel::result::Error),
+
+    #[error("Invalid IRI: {0}")]
+    Iri(#[from] iref::Error),
+
+    #[error("Json LD processing: {0}")]
```

Comment:
```suggestion
    #[error("JSON-LD processing: {0}")]
```
(may as well fix it if adjusting the line anyway)